app/src/main/java/com/example/haushaltsbuch/data/model/DevData.kt

```diff
@@ -1,55 +1,121 @@
 
-    var customers: MutableList<Customers>
-    var companies: MutableList<Company>
-    var subscriptionModel: MutableList<SubscriptionModel>
+    var customers: MutableList<Customer>? = null
+    var companies: MutableList<Company>? = null
+    var subscriptionModel: MutableList<SubscriptionModel>? = null
 
-    var expenses: MutableList<Expense>
-    var expenseCategories: MutableList<ExpenseCategory>
-    var retailers: MutableList<Retailer>
+    var expenses: MutableList<Expense>? = null
+    var expenseCategories: MutableList<Category>? = null
+    var retailers: MutableList<Merchant>? = null
 
-    var coupon: MutableList<Coupon>
-    var achievements: MutableList<Achievement>
+    var coupon: MutableList<Coupon>? = null
+    var achievements: MutableList<Achievement>? = null
 
-    var requests: MutableList<AppRequest>
-    var offers: MutableList<Offer>
-    var pricemodels: MutableList<PriceModel>
+    var requests: MutableList<Request>? = null
+    var offers: MutableList<Offer>? = null
+    var pricemodels: MutableList<PriceModel>? = null
 
-    var systemmessages: MutableList<Systemmessage>
     // transitive? friend hasRole hasAchieved
 
+    val addressCount = 50
+    val peopleCount = 50
+    val postcodeCount = 5
+    val streetcount = 10
+    val locationCount = 5
+
     fun generateData(){
-        people = MutableList<Person>(50, ::generatePerson)
+        locations = MutableList<Location>(locationCount, ::generateLocation)
+        postCode = MutableList<PostCode>(postcodeCount, ::generatePostCode)
+        street = MutableList<Street>(streetcount, ::generateStreet)
+        addresses = MutableList<Address>(50, ::generateAddress)
+        generateRoles()
+        users = MutableList(peopleCount, ::generateUser)
+        people = MutableList<Person>(peopleCount, ::generatePerson)
+
     }
 
-    fun generatePerson(): Person  {
+    fun generatePerson(index: Int): Person  {
         val name = faker.name.firstName()
         val surname = faker.name.lastName()
         val domain = faker.company.name()
         return Person(
-            name = name,
-            surname = surname,
-            email ="$name.$surname@$domain.de",
+                id = UUID.randomUUID(),
+                firstname = name,
+                lastname = surname,
+                email ="$name.$surname@$domain.de",
+                address = addresses!![index/(peopleCount/addressCount)],
+                user = users!![index],
+
     )}
 
-    fun generateAddresses()
+    fun generateAddress(index: Int): Address {
+        return Address(
+                postCode = postCode!![index/(addressCount/postcodeCount)],
+                street = street!![index/(addressCount/streetcount)],
+                person = null,
+        )
+    }
+
+    fun generatePostCode(index: Int): PostCode {
+        return PostCode(
+                postCode = faker.address.postcode(),
+                location = locations!![index/(postcodeCount/locationCount)],
```

Comment:
Ich bin mir nicht sicher warum du `[index/(postcodeCount/locationCount)]` verwendest, aber es könnten zu Problemen führen. Ich würde das so nicht im Code lassen.

Reply:
das ist genau, damit es nicht zu problemen führt. Die Anzahl an Postcodes und Locations könnten sich unterscheiden, damit dann nicht, falls es mehr postcodes als locations angelegt wird, ein zu hoher index verwendet wird, kommt bei der Formel höchstens der höchste index von location, aber vielleicht ist ja (index*locationcount/postcodeCount) verständlicher?

Reply:
Ja ich konnte jetzt nachvollziehen, warum du das so gemacht hast. Ich habe hier einen kleinen Verbesserungsvorschlag, der verhindert, dass Gleitkommazahlen als Index verwendet werden können. Mehr dazu in der PR